### PR TITLE
Adding nest hub translator, updating wink thermostat and nest thermos…

### PR DESCRIPTION
**Pull request by arjun-msft**
…tat to talk to their respective hub devices rather than helpers

This is not 100% ready to go, I have not updated tests and readme but I wanted code level feedback.

### CODE REVIEW

**Comment by arjun-msft** [comment]
The code aspect of this has been updated as well, please review.  I'd like to get this in.  I'll work on readme fixes and that should in theory be the last of it.

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+* This translator class implements the "Hub" interface.
+*/
+class Translator {
+    constructor(accessToken) {
+        this._accessToken = accessToken;
+
+        this._baseUrl = "https://developer-api.nest.com";
+        this._devicesPath = '/devices';
+
+        this._name = "Nest Hub"; // TODO: Can be pulled from OpenT2T global constants. This information is not available, at least, on wink hub.
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js]: >wink [](start = 135, length = 4)

nest*

File: org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/tests/test.js [hunk 2]
```diff
@@ -2,37 +2,63 @@ var test = require('ava');
...
 
 // Set/Get TargetTemperatureHigh via setters for individual properties
 test.serial('TargetTemperatureHigh', t => {
 
-    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
-        .then(translator => {
-            // TEST: translator is valid
-            t.is(typeof translator, 'object') && t.truthy(translator);
+    return OpenT2T.getPropertyAsync(translator, 'org.opent2t.sample.thermostat.superpopular', 'targetTemperatureHigh')
+        .then((targetTemperatureHigh2) => {
```
Comment (begillil) [inline org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/tests/test.js]: >targetTemperatureHigh2 [](start = 15, length = 22)

It's a little weird that targetTemperatureHigh2 comes before targetTemperatureHigh.

Consider using a more descriptive variable name.

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/manifest.xml [hunk 3]
```diff
@@ -12,9 +12,24 @@
       <schema id="oic.core" type="OCF" />
       <schema id="oic.baseresource" type="OCF" />
   </schemas>
-  <!-- associated onboarding module -->
   <onboarding id="org.opent2t.onboarding.nest">
```
Comment (begillil) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/manifest.xml]: ><onboarding id="org.opent2t.onboarding.nest"> [](start = 2, length = 45)

This whole section was removed from the Wink thermostat manifest. Why is it required for nest but not for Wink?

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            // var nestSmokeAlarms = nestDevices.smoke_co_alarms;
+
+            var filteredDevices = [];
+
+            // load thermostats
+            var nestThermostatIds = Object.keys(nestThermostats);
+            nestThermostatIds.forEach((nestThermostatId) => {
+                var nestDevice = nestThermostats[nestThermostatId];
+
+                // get the opent2t schema and translator for the wink device
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js]: >wink [](start = 65, length = 4)

nest

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                var opent2tInfo = this._getOpent2tInfo("thermostats");
+
+                // we only need to return certain properties back
+                var device = {};
+                device.name = nestDevice.name;
+
+                // set the specific device object id to be the id
+                device.id = nestThermostatId;
+
+                // set the opent2t info for the wink device
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js]: >wink [](start = 48, length = 4)

nest

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js (hunk 1 above)
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js]: And in several other comments)

---
In reply to: [78475948](https://github.com/openT2T/translators/pull/92#discussion_r78475948) [](ancestors = 78475948)

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+# Wink Hub translator
+Translator for Wink hub (https://wink.com)
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/README.md]: Any references to Wink should be updated to reference Nest

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "opent2t-translator-com-wink-hub",
+  "version": "1.0.0",
+  "description": "Wink Hub",
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/package.json]: Update to Nest

File: org.opent2t.sample.hub.superpopular/com.nest.hub/js/manifest.xml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest>
+  <schemas>
+      <schema id="org.opent2t.sample.hub.superpopular" main="true" type="OCF" />
+      <schema id="oic.core" type="OCF" />
+      <schema id="oic.baseResource" type="OCF" />
```
Comment (begillil) [inline org.opent2t.sample.hub.superpopular/com.nest.hub/js/manifest.xml]: nit: wrong level of indentation

File: org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/package.json [hunk 9]
```diff
@@ -19,12 +19,12 @@
   },
   "dependencies": {
     "opent2t": "1.0.1",
-    "opent2t-translator-helper-wink": "*"
+    "opent2t-translator-com-wink-hub": "*"
```
Comment (begillil) [inline org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/package.json]: There should be an equivalent change in the nest thermostat package.json, right?

**Comment by begillil** [comment]
:clock1:

**Comment by arjun-msft** [comment]
Thanks Ben for the review.  I've updated with your comments and all the other "soft" changes.

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/package.json [hunk 10]
```diff
@@ -20,13 +20,12 @@
   },
   "dependencies": {
     "opent2t": "1.0.1",
-    "opent2t-translator-helper-nest": "*"
+    "opent2t-translator-com-wink-hub": "*"
```
Comment (begillil) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/package.json]: >"opent2t-translator-com-wink-hub": "*" [](start = 4, length = 38)

These should depend on the nest hub, not wink.

**Pull request merged**